File: zena_mode/handlers/health.py
"""
Health Handler - Diagnostics, metrics, and status endpoints.
"""

import logging
from zena_mode.handlers.base import BaseZenHandler
from config_system import config

logger = logging.getLogger("ZenAI.Handler.Health")


class HealthHandler:
    """Namespace for health and diagnostics request handling."""

    @staticmethod
    def handle_get(handler: BaseZenHandler):
        """Routing for GET requests related to health/diagnostics."""

        if handler.path == "/health":
            from utils import is_port_active

            llm_status = is_port_active(config.llm_port)
            handler.send_json_response(
                200,
                {
                    "status": "healthy" if llm_status else "degraded",
                    "llm_online": llm_status,
                    "llm_port": config.llm_port,
                },
            )
            return True

        if handler.path == "/api/test-llm":
            from utils import is_port_active

            status = is_port_active(config.llm_port)
            handler.send_json_response(200, {"success": status, "port": config.llm_port})
            return True

        if handler.path == "/startup/progress":
            try:
                from startup_progress import get_startup_progress

                handler.send_json_response(200, get_startup_progress())
            except Exception:
                handler.send_json_response(200, {"stage": "loading", "percent": 0, "message": "Initializing..."})
            return True

        if handler.path == "/metrics":
            from zena_mode.profiler import monitor

            handler.send_json_response(200, monitor.get_summary())
            return True

        return False

File: rust_output/src/health.rs
/// Health Handler - Diagnostics, metrics, and status endpoints.

use anyhow::{Result, Context};
use crate::base::{BaseZenHandler};
use crate::config_system::{config};
use std::collections::HashMap;

pub static LOGGER: std::sync::LazyLock<String /* logging::getLogger */> = std::sync::LazyLock::new(|| Default::default());

/// Namespace for health and diagnostics request handling.
#[derive(Debug, Clone)]
pub struct HealthHandler {
}

impl HealthHandler {
    /// Routing for GET requests related to health/diagnostics.
    pub fn handle_get(handler: BaseZenHandler) -> Result<()> {
        // Routing for GET requests related to health/diagnostics.
        if handler.path == "/health".to_string() {
            // TODO: from utils import is_port_active
            let mut llm_status = is_port_active(config::llm_port);
            handler.send_json_response(200, HashMap::from([("status".to_string(), if llm_status { "healthy".to_string() } else { "degraded".to_string() }), ("llm_online".to_string(), llm_status), ("llm_port".to_string(), config::llm_port)]));
            true
        }
        if handler.path == "/api/test-llm".to_string() {
            // TODO: from utils import is_port_active
            let mut status = is_port_active(config::llm_port);
            handler.send_json_response(200, HashMap::from([("success".to_string(), status), ("port".to_string(), config::llm_port)]));
            true
        }
        if handler.path == "/startup/progress".to_string() {
            // try:
            {
                // TODO: from startup_progress import get_startup_progress
                handler.send_json_response(200, get_startup_progress());
            }
            // except Exception as _e:
            true
        }
        if handler.path == "/metrics".to_string() {
            // TODO: from zena_mode.profiler import monitor
            handler.send_json_response(200, monitor.get_summary());
            true
        }
        Ok(false)
    }
}
